### PR TITLE
Implement cart tracking

### PR DESCRIPTION
**Pull request by namenyi**
Story: https://app.clubhouse.io/recart/story/18749/implement-cart-tracking

### CODE REVIEW

File: lib/cart.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const cartItems = []
+    for (const lineItem of cart.lineItems) {
+      const quantity = Number(lineItem.quantity)
+      const price = Number(lineItem.variant.price)
+
+      // sometimes we can get it from the state, but it's a deprecated feature.
+      let product = products &&
+        products.products &&
+        products.products[lineItem.handle] &&
+        products.products[lineItem.handle].product
```
Comment (namenyi) [inline lib/cart.js]: We could add `lodash.get` to reduce the unnecessary complexity of the code

(or add a transpiler and use conditional chaining, but we agreed not to do that for now)

Comment (namenyi) [reply lib/cart.js]: And use it for all the `.variant.something` occurrences too

File: lib/recart-nacelle-plugin.js [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+  )
+}
+
+const initRecartPlugin = async (ctx) => {
   console.log('Recart Nacelle plugin started')
+
+  let timoutCounter = 0
+  while (!window._recart || !window._recart.isReady()) {
+    if (timoutCounter > 10) throw new Error('Recart is not initialized.')
+    await delay(1000)
```
Comment (namenyi) [inline lib/recart-nacelle-plugin.js]: I'd make the delay smaller. Waiting for a whole second would make the loading pretty slow.

How about `200` millisec delay and 50 retries?

File: lib/recart-nacelle-plugin.js [hunk 3]
```diff
@@ -1,7 +1,34 @@
...
+const delay = async milliseconds => {
+  return new Promise(resolve =>
+    setTimeout(() => resolve(true), milliseconds)
+  )
+}
+
+const initRecartPlugin = async (ctx) => {
   console.log('Recart Nacelle plugin started')
+
+  let timoutCounter = 0
```
Comment (namenyi) [inline lib/recart-nacelle-plugin.js]:
```suggestion
  let timeoutCounter = 0
```

Comment (namenyi) [reply lib/recart-nacelle-plugin.js]: `nit:` actually I'd find `retryCount` or `pollCount` a better name, but up to you

File: lib/recart-nacelle-plugin.js [hunk 4]
```diff
@@ -1,7 +1,34 @@
...
+  while (!window._recart || !window._recart.isReady()) {
+    if (timoutCounter > 10) throw new Error('Recart is not initialized.')
```
Comment (namenyi) [inline lib/recart-nacelle-plugin.js]: 🤔  should we really throw an error here? what is your line of thinking?

how about just logging an error and returning?

**Review by gabor-recart — APPROVED** [review]

**Pull request merged**